src/newmatrix.cpp

```diff
@@ -3110,6 +3065,75 @@ void matrix_update()
   }
 }
 
+// For things that should happen no more than once an hour.
+void matrix_hour_update()
+{
+  PERF_PROF_SCOPE(pr_, __func__);
+  rnum_t rnum = 1;
+  extern struct time_info_data time_info;
+  for (;rnum <= top_of_matrix; rnum++) {
+    bool decker = FALSE;
+
+    // Paydara resets at midnight
+    if (time_info.hours == 0) {
+      if (HOST.type == HOST_DATASTORE && HOST.undiscovered_paydata <= 0) {
+        reset_host_paydata(rnum);
+      }
+    }
+
+    // Host alert state and subsystem encryption reviewed every 2 hours.
+    // But only if no deckers in the area.
+    if (time_info.hours % 2 == 0) {
+      // Check the current host.
+      for (struct matrix_icon *icon = HOST.icons; icon; icon = icon->next_in_host) {
+        if (!ICON_IS_IC(icon)) {
+          decker = TRUE;
+          break;
+        }
+      }
+      // Check surrounding hosts.
+      struct host_data *adjacent_host = NULL;
+      for (struct entrance_data *entrance = HOST.entrance; entrance && !decker; entrance = entrance->next) {
+        if ((adjacent_host = entrance->host)) {
+          for (struct matrix_icon *icon = adjacent_host->icons; icon; icon = icon->next_in_host) {
+            if (!ICON_IS_IC(icon)) {
+              decker = TRUE;
+              break;
+            }
+          }
+        }
+      }
+
+      // No deckers? Do the things.
+      if (!decker) {
+        // Alert active(2) to passive(1), passive to none(0).
+        // Note: alert = 3 means host has been shut down, handled in matrix_update.
+        if ((HOST.alert > 0) && (HOST.alert < 3)) {
+          HOST.alert--;
+        }
+
+        // Re-encrypt subsystems.
+        for (int x = ACCESS; x < NUM_OF_SUBSYSTEMS; x++) {
+          if (HOST.stats[x][MTX_STAT_ENCRYPTED] != 0 && HOST.stats[x][MTX_STAT_ENCRYPTED] != 1) {
+            char warnbuf[1000];
+            snprintf(warnbuf, sizeof(warnbuf), "WARNING: %s mtx_stat_encrypted on %ld is %ld (must be 1 or 0)!",
+                      acifs_strings[x],
+                      HOST.vnum,
+                      HOST.stats[x][MTX_STAT_ENCRYPTED]);
+            mudlog(warnbuf, NULL, LOG_SYSLOG, TRUE);
```

Comment:
No need to change, but FYI this is old-style logging. You can now use `mudlog_vfprintf(NULL, LOG_SYSLOG, "message", var1, var2 ...);` and skip the declaration of a buffer and sprintf() call.